qp_klp/Metagenomic.py

```diff
@@ -0,0 +1,161 @@
+from os import walk
+from os.path import exists, join, basename
+from sequence_processing_pipeline.PipelineError import PipelineError
+import pandas as pd
+from qp_klp.Step import Step, FailedSamplesRecord
+
+
+class Metagenomic(Step):
+    def __init__(self, pipeline, master_qiita_job_id, sn_tid_map_by_project,
+                 status_update_callback=None):
+        super().__init__(pipeline,
+                         master_qiita_job_id,
+                         sn_tid_map_by_project,
+                         status_update_callback)
+
+        if pipeline.pipeline_type != 'metagenomic':
```

Comment:
Should this be metatranscriptomic too? Note that this might be concerning as there are several places where this check is done, perhaps having a global set or something might be better; it could be part of the object too.